service/history/workflow/mutable_state_impl.go

```diff
@@ -275,8 +280,12 @@ func NewMutableState(
 		ExecutionStats:   &persistencespb.ExecutionStats{HistorySize: 0},
 	}
 	s.approximateSize += s.executionInfo.Size()
-	s.executionState = &persistencespb.WorkflowExecutionState{State: enumsspb.WORKFLOW_EXECUTION_STATE_CREATED,
-		Status: enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING}
+	s.executionState = &persistencespb.WorkflowExecutionState{
+		RunId: runID,
```

Comment:
Can you remove the current logic for populating namespaceID, workflowID and runID? or you can do a sanity check there to see they match or not.

Reply:
removing current setters and also doing sanity check